version_updates.md

```diff
@@ -0,0 +1,130 @@
+
+When a new set of Operators is installed via OLM (either manually or automatically)
+the OpenStackVersion resources will reconcile and a new 'AvailableVersion' will
+be set on each CR.
+
+In order to apply an update an administrator sets the 'TargetVersion' to the
+same value as the 'AvailableVersion' on the OpenStackVersion resources. This
+triggers the OpenStackVersion to reconcile and will update the images
+on the associated OpenStackControlplane and OpenStackDataplane resources in
+that namespace.
+
+```shell
+$ oc get openstackversion
+NAME              TARGET VERSION   AVAILABLE VERSION
+openstack         1.0.0            1.0.1
+```
+
+```yaml
+apiVersion: core.openstack.org/v1beta1
+kind: OpenStackVersion
+metadata:
+  labels:
+  name: openstack
+spec:
+  openstackControlPlaneName: openstack
+  targetVersion: 1.0.1
+```
+
+## Custom images for Cinder Volume
+
+In order to allow administrators to configure customized Cinder volume backends
+with custom containerImages for each backend you must specify those images
+from automatic updates. This is because there is no way to fully automate
+the creation of those images at this time. To configure a custom
+Cinder volume named 'netapp' updates you can add the name 'netapp' to the
+'cinderVolumeCustomImages' map on the OpenStackVersion CR and configure
+a custom image for that volume backend. Multiple named Cinder volume backends may
+be configured in this manner.
+
+```yaml
+apiVersion: core.openstack.org/v1beta1
+kind: OpenStackVersion
+metadata:
+  labels:
+  name: openstack
+spec:
+  openstackControlPlaneName: openstack
+  targetVersion: 1.0.1
+  cinderVolumeCustomImages:
+    netapp: <custom image URL location>
+```
+
+## Custom Images for Neutron
+
+TBD
+
+## Custom images for other OpenStack services
+
+In general administrators should not need to customize containerImages
+for any other OpenStack services images. The exception would be
+the need to apply a hotfix. In order to accommodate a hotfixed or
+patched containerImage you can update the 'serviceCustomContainerImages'
+map on the OpenStackVersion to contain the camel cased
+name for the services that needs to be customized
+with a custom container image. For example if you need
+to hotfix the Glance API service container image you could
+add 'GlanceApi' and the custom container image to the
+'serviceCustomContainerImages' map.
+
+```yaml
+apiVersion: core.openstack.org/v1beta1
+kind: OpenStackVersion
+metadata:
+  labels:
+  name: openstack
+spec:
+  openstackControlPlaneName: openstack
+  targetVersion: 1.0.1
+  serviceCustomContainerImages:
+    GlanceApi: <custom image URL location>
+```
+
+## Custom ordering of service updates
+
+Container image updates currently occur in parallel for a given CR.
+First OVN will be updated on the Dataplane CRs. Then the Controlplane will be updated.
+Finally the rest of the Dataplane CRs container images will be updated.
+
+If a situation arises where we need to update one service before
```

Comment:
I think the situation just arised (see ovnController remark).

Reply:
It is well understood that both the OVN controllers on the dataplane and controlplane need to be updated for our initial minor updates support. This comment was really about future unforeseen situations that may arise here. I hope this makes sense...